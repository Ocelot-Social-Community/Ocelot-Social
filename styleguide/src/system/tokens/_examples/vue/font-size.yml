#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
