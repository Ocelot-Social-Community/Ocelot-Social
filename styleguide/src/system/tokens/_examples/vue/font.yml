#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
