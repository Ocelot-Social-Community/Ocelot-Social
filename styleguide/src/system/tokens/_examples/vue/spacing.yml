#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
