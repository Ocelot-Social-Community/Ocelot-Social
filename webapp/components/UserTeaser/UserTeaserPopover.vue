<template>
  <div class="user-teaser-popover">
    <badges
      v-if="$env.BADGES_ENABLED"
      :badges="[user.badgeVerification, ...user.badgeTrophiesSelected]"
    />
    <location-info
      v-if="user.location"
      :location="{ ...user.location, distanceToMe: user.distanceToMe }"
      class="location-info"
    />
    <ul class="statistics">
      <li>
        <ds-number :count="user.followedByCount" :label="$t('profile.followers')" />
      </li>
      <li>
        <ds-number
          :count="user.contributionsCount"
          :label="$t('common.post', null, user.contributionsCount)"
        />
      </li>
      <li>
        <ds-number
          :count="user.commentedCount"
          :label="$t('common.comment', null, user.commentedCount)"
        />
      </li>
    </ul>
    <nuxt-link v-if="isTouchDevice && userLink" :to="userLink" class="link">
      <ds-button primary>{{ $t('user-teaser.popover.open-profile') }}</ds-button>
    </nuxt-link>
  </div>
</template>

<script>
import Badges from '~/components/Badges.vue'
import LocationInfo from '~/components/UserTeaser/LocationInfo.vue'
import { isTouchDevice } from '~/components/utils/isTouchDevice'

export default {
  name: 'UserTeaserPopover',
  components: {
    Badges,
    LocationInfo,
  },
  props: {
    user: { type: Object, default: null },
    userLink: { type: Object },
  },
  computed: {
    isTouchDevice() {
      return isTouchDevice()
    },
  },
}
</script>

<style scoped>
.user-teaser-popover {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 16px;
  gap: 16px;
  min-width: 200px;
}

.location-info {
  margin-bottom: 12px;
}

.link {
  margin-top: 16px;
}

.statistics {
  display: flex;
  justify-content: space-between;
  width: 100%;
}
</style>
