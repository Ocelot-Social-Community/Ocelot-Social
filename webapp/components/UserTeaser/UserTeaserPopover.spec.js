import { render } from '@testing-library/vue'
import { RouterLinkStub } from '@vue/test-utils'
import UserTeaserPopover from './UserTeaserPopover.vue'

const localVue = global.localVue

const user = {
  id: 'id',
  name: 'Tilda Swinton',
  slug: 'tilda-swinton',
  badgeVerification: {
    id: 'bv1',
    icon: '/icons/verified',
    description: 'Verified',
    isDefault: false,
  },
  badgeTrophiesSelected: [
    {
      id: 'trophy1',
      icon: '/icons/trophy1',
      description: 'Trophy 1',
      isDefault: false,
    },
    {
      id: 'trophy2',
      icon: '/icons/trophy2',
      description: 'Trophy 2',
      isDefault: false,
    },
    {
      id: 'empty',
      icon: '/icons/empty',
      description: 'Empty',
      isDefault: true,
    },
  ],
}

const userLink = {
  name: 'profile-id-slug',
  params: { slug: 'slug', id: 'id' },
}

describe('UserTeaserPopover', () => {
  const Wrapper = ({ badgesEnabled = true, withUserLink = true, onTouchScreen = false }) => {
    const mockIsTouchDevice = onTouchScreen
    jest.mock('../utils/isTouchDevice', () => ({
      isTouchDevice: jest.fn(() => mockIsTouchDevice),
    }))
    return render(UserTeaserPopover, {
      localVue,
      propsData: {
        userId: 'id',
        userLink: withUserLink ? userLink : null,
      },
      data: () => ({
        User: [user],
      }),
      stubs: {
        NuxtLink: RouterLinkStub,
      },
      mocks: {
        $t: jest.fn((t) => t),
        $env: {
          BADGES_ENABLED: badgesEnabled,
        },
      },
    })
  }

  describe('given a touch device', () => {
    it('shows button when userLink is provided', () => {
      const wrapper = Wrapper({ withUserLink: true, onTouchScreen: true })
      expect(wrapper.container).toMatchSnapshot()
    })

    it('does not show button when userLink is not provided', () => {
      const wrapper = Wrapper({ withUserLink: false, onTouchScreen: true })
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  describe('given a non-touch device', () => {
    it('does not show button when userLink is provided', () => {
      const wrapper = Wrapper({ withUserLink: true, onTouchScreen: false })
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  it('shows badges when enabled', () => {
    const wrapper = Wrapper({ badgesEnabled: true })
    expect(wrapper.container).toMatchSnapshot()
  })

  it('does not show badges when disabled', () => {
    const wrapper = Wrapper({ badgesEnabled: false })
    expect(wrapper.container).toMatchSnapshot()
  })
})
