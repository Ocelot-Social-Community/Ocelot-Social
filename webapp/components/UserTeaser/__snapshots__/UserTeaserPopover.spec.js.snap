// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTeaserPopover does not show badges when disabled 1`] = `
<div>
  <div
    class="user-teaser-popover"
  >
    <!---->
     
    <!---->
     
    <ul
      class="statistics"
    >
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    profile.followers
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.post
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.comment
  
          </p>
        </div>
      </li>
    </ul>
     
    <!---->
  </div>
</div>
`;

exports[`UserTeaserPopover given a non-touch device does not show button when userLink is provided 1`] = `
<div>
  <div
    class="user-teaser-popover"
  >
    <div
      class="hc-badges"
    >
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/verified"
          title="Verified"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy1"
          title="Trophy 1"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy2"
          title="Trophy 2"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/empty"
          title="Empty"
        />
      </div>
    </div>
     
    <!---->
     
    <ul
      class="statistics"
    >
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    profile.followers
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.post
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.comment
  
          </p>
        </div>
      </li>
    </ul>
     
    <!---->
  </div>
</div>
`;

exports[`UserTeaserPopover given a touch device does not show button when userLink is not provided 1`] = `
<div>
  <div
    class="user-teaser-popover"
  >
    <div
      class="hc-badges"
    >
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/verified"
          title="Verified"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy1"
          title="Trophy 1"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy2"
          title="Trophy 2"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/empty"
          title="Empty"
        />
      </div>
    </div>
     
    <!---->
     
    <ul
      class="statistics"
    >
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    profile.followers
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.post
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.comment
  
          </p>
        </div>
      </li>
    </ul>
     
    <!---->
  </div>
</div>
`;

exports[`UserTeaserPopover given a touch device shows button when userLink is provided 1`] = `
<div>
  <div
    class="user-teaser-popover"
  >
    <div
      class="hc-badges"
    >
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/verified"
          title="Verified"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy1"
          title="Trophy 1"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy2"
          title="Trophy 2"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/empty"
          title="Empty"
        />
      </div>
    </div>
     
    <!---->
     
    <ul
      class="statistics"
    >
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    profile.followers
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.post
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.comment
  
          </p>
        </div>
      </li>
    </ul>
     
    <!---->
  </div>
</div>
`;

exports[`UserTeaserPopover shows badges when enabled 1`] = `
<div>
  <div
    class="user-teaser-popover"
  >
    <div
      class="hc-badges"
    >
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/verified"
          title="Verified"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy1"
          title="Trophy 1"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/trophy2"
          title="Trophy 2"
        />
      </div>
      <div
        class="hc-badge-container"
      >
        <img
          class="hc-badge"
          src="/api/icons/empty"
          title="Empty"
        />
      </div>
    </div>
     
    <!---->
     
    <ul
      class="statistics"
    >
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    profile.followers
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.post
  
          </p>
        </div>
      </li>
       
      <li>
        <div
          class="ds-number ds-number-size-x-large"
        >
          <p
            class="ds-text ds-number-count ds-text-size-x-large"
            style="margin-bottom: 0px;"
          >
            0
          </p>
          <p
            class="ds-text ds-number-label ds-text-size-small ds-text-soft"
          >
            
    common.comment
  
          </p>
        </div>
      </li>
    </ul>
     
    <!---->
  </div>
</div>
`;
